frontend/src/components/playlist.js

```diff
@@ -1,19 +1,254 @@
+        (err) => {
+            console.log(err)
+        }
+    )
+
+    this.setState({
+        topTracksIds: []
+    })
+  }
+
+
+  createPlaylist = (userId, name, description) => {
+      spotifyApi.createPlaylist(userId, {name, description}).then(
+        (data) => {
+
+            console.log('once' + this.state.user.id)
+            this.setState({
+                user: { id: userId ,playlistId: data.id, iframe: `https://open.spotify.com/embed/playlist/${data.id}` } 
+            }, () => {
+                this.searchArtist(this.state.selectedArtists)
+            })
+            
+        },
+        (err) => {
+            console.log(err)
+        }
+      )
+  }
+
+
+  handleSelect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.concat(value)
+    }, () => {
+        console.log(this.state.selectedArtists)
+    }) 
+  }
+
+  handleDeselect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.filter(name => !name.includes(value))
+    }, () => {
+        console.log(this.state.selectedArtists)
+    });
+
+
+  }
+
+  handleCreate = () => {
+    this.createPlaylist(this.state.user.id, this.state.playlistName, this.state.playlistDescription);
+    this.setState({
+        displayPlaylist: 'block',
+      })
+
+    setTimeout(() => { // iframe needs a reload to show songs
+        this.setState({ random: this.state.random + 1 })
+    }, 700);
+
+    setTimeout(() => { // selected songs dissapear 
+        this.setState({
+            selectedArtists: [],
+            topTracksIds: []
+        })
+    },3000)
+  }
+
+  handleNameInput = (event) => {
+    this.setState({
+        playlistName: event.target.value
+    })
+  }
+
+  handleDescriptionInput = (event) => {
+    this.setState({
+        playlistDescription: event.target.value
+    })
+  }
+
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1>
-            PLAYLIST PAGE
-          </h1>
-        </header>
-      </div>
-    );
+        <Container className="playlistPage-container">
+            {this.state.displayPlaylist !== 'none' ? <Row id="iframe-row"><Embedded key={this.state.random} link={this.state.user.iframe}/></Row> : ''}
+            <Row id="playlistPage-row">
+                <Col>
+                    <div id="form-container">
+                        <div id="artistInput">
+                            <div id="searchArtist">
+                                <Search
+                                    id="searchArtistInput"
+                                    placeholder="Search Artist"
+                                    onSearch={ value => this.getArtist(value)}
+                                />
+                            </div>
+                            <Button id ="create-button" type="primary" onClick={this.handleCreate}> Create Playlist </Button>
```

Comment:
Can we hide this until the user has input all of the information. When I first used this I wasnt sure at what stage I was allowed to create a playlist. We should make it clear to the user when they are able to create the playlist.